embassy-stm32/src/crs.rs

```diff
@@ -0,0 +1,67 @@
+use embassy_hal_common::PeripheralRef;
+
+use crate::pac::CRS as PAC_CRS;
+use crate::peripherals::CRS;
+use crate::rcc::enable_hsi48;
+use crate::rcc::sealed::RccPeripheral;
+use crate::{pac, Peripheral};
+
+pub struct Config {
+    sync_source: SyncSource,
+}
+
+pub enum SyncSource {
+    Gpio,
+    Lse,
+    Usb,
+}
+
+pub struct Crs<'d> {
+    _inner: PeripheralRef<'d, CRS>,
+    _config: Config,
+}
+
+impl Config {
+    #[inline]
+    pub fn default() -> Config {
+        Config {
+            sync_source: SyncSource::Usb,
+        }
+    }
+}
+
+impl<'d> Crs<'d> {
+    pub fn new(peripheral: impl Peripheral<P = CRS> + 'd, config: Config) -> Self {
+        unsafe {
+            enable_hsi48();
+        }
+
+        CRS::enable();
+
+        match config.sync_source {
+            SyncSource::Gpio => {
+                unimplemented!("GPIO CRS Sync Source is not supported");
```

Comment:
I don't have a good setup to test these but I can at least set the source bits properly.